File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        i=0
        j=len(skill)-1
        t=skill[i] + skill[j]
        pro = 0
        while(i<j):
            if(skill[i] + skill[j] == t):
                pro += skill[i] * skill[j]
            else:
                return -1
            i+=1
            j-=1
        return pro

File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(), skill.end());
        int n = skill.size();
        int i=0, j=n-1;
        int t = skill[i] + skill[j];
        long long pro = 0;
        while(i<j){
            if(skill[i] + skill[j] == t){
                pro += skill[i]*skill[j];
            }
            else{
                return -1;
            }
            i++; j--;
        }
        return pro;
    }
};